File: core/src/main/java/pcb/uwu/unit/derived/optics/IlluminanceUnit.java
package pcb.uwu.unit.derived.optics;

import pcb.uwu.core.CompositeUnit;
import pcb.uwu.core.UnitCounter;
import pcb.uwu.unit.base.LengthUnit;
import pcb.uwu.unit.base.LuminousIntensityUnit;
import pcb.uwu.unit.derived.area.AreaUnit;
import pcb.uwu.unit.derived.scalar.SolidAngleUnit;

public class IlluminanceUnit extends CompositeUnit {

	// region base constructor

	public IlluminanceUnit(LuminousIntensityUnit luminousIntensityUnit, SolidAngleUnit solidAngleUnit, LengthUnit lengthUnit) {
		super(new UnitCounter()
				.major(luminousIntensityUnit.getUnitCounter())
				.major(solidAngleUnit)
				.minor(lengthUnit, 2));
	}

	public IlluminanceUnit(LuminousFluxUnit luminousFluxUnit, AreaUnit areaUnit) {
		super(new UnitCounter()
				.major(luminousFluxUnit.getUnitCounter())
				.minor(areaUnit.getUnitCounter()));
	}

	// endregion

}

File: core/src/main/java/pcb/uwu/unit/derived/optics/IlluminanceUnit.kt
package pcb.uwu.unit.derived.optics

import pcb.uwu.core.CompositeUnit
import pcb.uwu.core.UnitCounter
import pcb.uwu.unit.base.LengthUnit
import pcb.uwu.unit.base.LuminousIntensityUnit
import pcb.uwu.unit.derived.fundamental.AreaUnit
import pcb.uwu.unit.scalar.SolidAngleUnit

open class IlluminanceUnit : CompositeUnit
{
    constructor(luminousIntensityUnit: LuminousIntensityUnit,
                solidAngleUnit: SolidAngleUnit,
                lengthUnit: LengthUnit)
            : super(UnitCounter()
                        .major(luminousIntensityUnit.unitCounter)
                        .major(solidAngleUnit)
                        .minor(lengthUnit, 2))

    constructor(luminousFluxUnit: LuminousFluxUnit,
                areaUnit: AreaUnit)
            : super(UnitCounter()
                        .major(luminousFluxUnit.unitCounter)
                        .minor(areaUnit.unitCounter))
}